src/Nethermind/Nethermind.Monitoring/Metrics/MetricsController.cs

```diff
@@ -213,83 +369,25 @@ private void UpdateMetrics(Type type)
         {
             EnsurePropertiesCached(type);
 
-            foreach ((MemberInfo _, string gaugeName, Func<double> accessor) in _membersCache[type])
+            foreach (IMetricUpdater metricUpdater in _metricUpdaters[type])
             {
-                ReplaceValueIfChanged(accessor(), gaugeName);
+                metricUpdater.Update();
             }
+        }
 
-            foreach (DictionaryMetricInfo info in _dictionaryCache[type])
+        private static Func<double> GetValueAccessor(MemberInfo member)
```

Comment:
`MemberExtensions` class with signature:

```
        private static Func<T> GetValueAccessor<T>(MemberInfo member)
```

Also consider using internally `Delegate.CreateDelegate` instead of `GetValue`. Not sure if still in dotnet 9, but in earlier it was way faster.

Reply:
Also:
```
        private static Func<double> GetValueAccessor(MemberInfo member) =>
            member switch
            {
                PropertyInfo property => () => Convert.ToDouble(property.GetValue(null)),
                FieldInfo field => () => Convert.ToDouble(field.GetValue(null)),
                _ => throw new InvalidOperationException($"Type of {member} is not handled")
            };

```